src/main/java/gov/usgs/ngwmn/logic/WaterLevelStatistics.java

```diff
@@ -173,19 +144,67 @@ public JsonData calculate(Specifier spec, List<WLSample> samples) {
 		
 		return builder.build();
 	}
-	
 
-	protected void overallStats(List<WLSample> samples, List<WLSample> sortedByValue) {
-		overallStatistics.overallStats(samples, sortedByValue);
-		if (samples == null || samples.size() == 0) {
+
+	protected void overallLastestPercentile(List<WLSample> samplesByDate) {
+		// get the latest (most recent) sample
+		int last = samplesByDate.size()-1;
+		WLSample latestSample = samplesByDate.get(last);
+		// get the sample for the same month as the latest sample
+		String month = Value.padMonth(Value.monthUTC(latestSample.time));
+		List<WLSample> monthSamples = monthlyStats.filterValuesByGivenMonth(samplesByDate, month);
+		// get the medians for each year-month
+		List<WLSample> normalizeMutlipleYearlyValues = 
+				monthlyStats.medianMonthlyValues(monthSamples,  monthlyStats.sortFunctionByQualifier());
+		// the most recent must now be added into the collection and replace of the year-month it represents
+		replaceLatestSample(normalizeMutlipleYearlyValues, latestSample);
+		// get the percentile of the latest sample
+		BigDecimal latestPercentile = percentileOfValue(normalizeMutlipleYearlyValues, latestSample, Value::valueOf);
+		builder.latestPercentile(latestPercentile.toString());
+	}
+
+	protected void replaceLatestSample(List<WLSample> normalizeMutlipleYearlyValues, WLSample latestSample) {
+		if (normalizeMutlipleYearlyValues.contains(latestSample)) {
+			return; // if it happens to be in there leave it be
+		}
+		// initial conditions and values
+		int        compareToDir  = MediationType.BelowLand.equal(builder.mediation()) ?1 :-1;
```

Comment:
What does `Dir` on `compareToDir` stand for?

Reply:
Ah, direction, saw it a few lines later.  Can we change it to `compareToDirection`?

Reply:
Also, is the `.equal` method our own comparison method or should it be `.equals`?

Reply:
I change the name of both. eqaulSortOrder and compareToVal are the new names.